src/ui/Distance/DistanceResources.ts

```diff
@@ -0,0 +1,245 @@
+import {
+  DistanceEvents,
+  IResolvingPositionEventArgs,
+  IPosition,
+  IPositionProvider,
+  IPositionResolvedEventArgs
+} from '../../events/DistanceEvents';
+
+import { Component } from '../Base/Component';
+import { ComponentOptions } from '../Base/ComponentOptions';
+import { IComponentBindings } from '../Base/ComponentBindings';
+import { InitializationEvents, QueryEvents } from '../../EventsModules';
+import { IQueryFunction } from '../../rest/QueryFunction';
+import { IBuildingQueryEventArgs } from '../../events/QueryEvents';
+import { Dom } from '../../UtilsModules';
+import { Initialization } from '../Base/Initialization';
+import { get } from '../Base/RegisteredNamedMethods';
+import { exportGlobally } from '../../GlobalExports';
+
+export interface IDistanceOptions {
+  distanceField: string;
+  latitudeField: string;
+  longitudeField: string;
+  unitConversionFactor: number;
+  disabledDistanceCssClass: string;
+}
+
+/**
+ * The `DistanceResources` component defines a field that computes the distance according to the user's position.
+ *
+ * You must use a position provider to provide a position to this component.
+ * 
+ * Components that uses the current distance should be disabled until a distance is provided by this component.
+ * 
+ * See also [`DistanceEvents`]{@link DistanceEvents} for events triggered by this component, 
+ *
+ * See also the components that can provide a position.
+ * 
+ * * [`GoogleApiPositionProvider`]{@link GoogleApiPositionProvider}
+ * * [`NavigatorPositionProvider`]{@link NavigatorPositionProvider}
+ * * [`StaticPositionProvider`]{@link StaticPositionProvider}
+ */
+export class DistanceResources extends Component {
+  public static ID = 'DistanceResources';
+  static doExport = () => {
+    exportGlobally({
+      DistanceResources: DistanceResources
+    });
+  };
+  private latitude: number;
+  private longitude: number;
+
+  private lastPositionRequest: Promise<IPosition | void>;
+
+  /**
+   * The possible options for a DistanceResources.
+   * @componentOptions
+   */
+  public static options: IDistanceOptions = {
+    /**
+     * Specifies the field that will contain the distance value.
+     *
+     * Specifying a value for this option is required for the `DistanceResources` component to work.
+     */
+    distanceField: ComponentOptions.buildStringOption({
+      required: true
+    }),
+    /**
+     * Specifies the field that contains the latitude value.
+     *
+     * Specifying a value for this option is required for the `DistanceResources` component to work.
+     */
+    latitudeField: ComponentOptions.buildStringOption({
```

Comment:
latitudeField should be an `IFieldOption` and you can use `ComponentOptions.buildFieldOption`